src/panel/graph/AudioGraphRender.ts

```diff
@@ -77,22 +85,29 @@ export class AudioGraphRender {
     this.camera.viewportObserver.observe((viewport) => {
       const {x, y, width, height} = this.camera.viewport;
       app.stage.setTransform(-x / width, -y / height, 1 / width, 1 / height);
+      this.requestRender();
     });
   }
 
   /** Render the graph. */
-  render() {
-    const {pixiApplication: app} = this;
+  requestRender() {
+    if (this.renderFrameId === null) {
+      this.renderFrameId = requestAnimationFrame(this._render);
+    }
+  }
+
+  _render() {
+    this.renderFrameId = null;
 
-    this.renderFrameId = requestAnimationFrame(this.render);
+    const {pixiApplication: app} = this;
 
     this.camera.setScreenSize(app.screen.width, app.screen.height);
     app.render();
   }
 
   /** Start rendering regularly. */
   start() {
-    this.renderFrameId = requestAnimationFrame(this.render);
+    // this.renderFrameId = requestAnimationFrame(this.render);
```

Comment:
This line is commented. Is this intentional?

Reply:
It is intentional. Additionally I meant to remove `start`. It is replaced by `requestRender` in this changeset. `requestRender` is called when something has changed and the canvas rendering should be updated. This way the canvas only renders when the graph changes or the user pans or zooms the canvas.